bundles/core/org.eclipse.smarthome.core.audio/src/main/java/org/eclipse/smarthome/core/audio/internal/AudioServlet.java

```diff
@@ -0,0 +1,229 @@
+import org.eclipse.smarthome.core.audio.AudioHTTPServer;
+import org.eclipse.smarthome.core.audio.AudioStream;
+import org.osgi.framework.BundleContext;
+import org.osgi.service.http.HttpContext;
+import org.osgi.service.http.HttpService;
+import org.osgi.service.http.NamespaceException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A servlet that serves audio streams via HTTP.
+ *
+ * @author Kai Kreuzer - Initial contribution and API
+ *
+ */
+public class AudioServlet extends HttpServlet implements AudioHTTPServer {
+
+    private static final long serialVersionUID = -3364664035854567854L;
+
+    private static final String SERVLET_NAME = "/audio";
+
+    private final Logger logger = LoggerFactory.getLogger(AudioServlet.class);
+
+    private Map<String, AudioStream> streams = new ConcurrentHashMap<>();
+
+    protected HttpService httpService;
+    private BundleContext bundleContext;
+
+    protected void activate(BundleContext bundleContext) {
+        this.bundleContext = bundleContext;
+    }
+
+    protected void deactivate(BundleContext bundleContext) {
+        this.bundleContext = null;
+    }
+
+    protected void setHttpService(HttpService httpService) {
+        this.httpService = httpService;
+
+        try {
+            logger.debug("Starting up the audio servlet at " + SERVLET_NAME);
+            Hashtable<String, String> props = new Hashtable<String, String>();
+            httpService.registerServlet(SERVLET_NAME, this, props, createHttpContext());
+        } catch (NamespaceException e) {
+            logger.error("Error during servlet startup", e);
+        } catch (ServletException e) {
+            logger.error("Error during servlet startup", e);
+        }
+    }
+
+    protected void unsetHttpService(HttpService httpService) {
+        httpService.unregister(SERVLET_NAME);
+        this.httpService = null;
+    }
+
+    /**
+     * Creates an {@link HttpContext}.
+     *
+     * @return an {@link HttpContext} that grants anonymous access
+     */
+    protected HttpContext createHttpContext() {
+        // TODO: Once we have a role-based permission system in place, we need to make sure that we create an
+        // HttpContext here, which allows accessing the servlet without any authentication.
+        HttpContext httpContext = httpService.createDefaultHttpContext();
+        return httpContext;
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        String streamId = StringUtils.substringAfterLast(req.getRequestURI(), "/");
+        if (!streams.containsKey(streamId)) {
+            logger.debug("Received request for invalid stream id at  {}", req.getRequestURI());
+            resp.sendError(HttpServletResponse.SC_NOT_FOUND);
+        } else {
+            logger.debug("Stream to serve is {}", streamId);
+
+            AudioStream stream = streams.get(streamId);
```

Comment:
Definitely yes! Good catch, I only changed that while debugging and missed to replace it again.